### DISCUSSION TITLE
Unnecessary stubbings detected - false positive?

### DISCUSSION THREAD

**Discussion by rsteppac** [mockk/mockk#963] [Q&A]
I have a problem with what appears to be a false positive for unnecessary stubbing. This is MockK version `1.13.2`.

All my Junit 5 tests are annotated as 

```java
@MockKExtension.CheckUnnecessaryStub
@ExtendWith(MockKExtension::class)
```

Two of my test classes seem to interact in a "static" way in that test class `A` creates a mock instance of `X` (`MessageContext`) and stubs a certain method (`getResponse()`). This is the actual (abridged) code:

```java
@MockK(relaxed = true)
private lateinit var responseMessage: SaajSoapMessage

@MockK(relaxUnitFun = true)
private lateinit var messageContext: MessageContext

@BeforeEach
fun setUp() {
    every { messageContext.response } returns responseMessage
}
```

And test class `B`, which does neither create a mock of type X, nor the real thing, and neither does the code under test, fails with _Unnecessary stubbings detected_ for the method stubbed in `A`.

> Unnecessary stubbings detected.
> Following stubbings are not used, either because there are unnecessary or because tested code doesn't call them :
> 
> 1) MessageContext(messageContext#12).getResponse())
> java.lang.AssertionError: Unnecessary stubbings detected.

As soon as I annotate `A` with `@Disabled` the tests in class `B` run through without a problem and pass the check for unnecessary stubbing.

How can I properly resolve this, i.e. not having to disable the check for unnecessary stubs?

**Comment by rsteppac** [comment]
Instead of disabling the check for unnecessary stubbings I moved the execution of class `A` to the last position as a workaround with the help of Junit's `org.junit.jupiter.api.Order` annotation. Using the annotation to execute `A` as the very first test class I discovered that this will make **all** my test classes fail with '_unncessary stubbing_' errors. Not just one other test class. 
The only thing that I am doing in `A` that I do not do, in principal, in the other test classes, is mock a top level function like so: 
```kotlin
    @BeforeEach
    fun setUp() {
        mockkStatic(::extractUserDetails)
        //...
    }
```
But even if I remove the that static mock the behavior stays the same.

Any advice on this would be greatly appreciated.

**Comment by eduard-onofrei** [comment]
The same thing is happening to me

**Comment by JeremiahDo** [comment]
I am facing the same behaviour.

**Comment by mpedroc90-pleo** [comment]
I’m encountering flaky tests when combining @ExtendWith(MockKExtension::class) with `mockk()` function. The flakiness appears when I run the entire test suite in parallel.

My initial suspicion is that the cleanup process isn’t happening correctly.
